scripts/updateOrderPortal.py

```diff
@@ -1,41 +1,111 @@
+        pjs = self.lims.get_projects(open_date=open_date.strftime("%Y-%m-%d"))
+        for project in pjs:
+            if project.open_date:
+                project_ngi_identifier=project.id
+                project_ngi_name=project.name
+                try:
+                    ORDER_ID=project.udf['Portal ID']
+                except KeyError:
+                    continue
+
+                url = "{base}/api/v1/order/{id}".format(base=self.base_url, id=ORDER_ID)
+                data = {'fields': {'project_ngi_name': project.name, 'project_ngi_identifier': project.id}}
+                if not dry_run:
+                    response = requests.post(url, headers=self.headers, json=data)
+                    assert response.status_code == 200, (response.status_code, response.reason)
 
-    lims = Lims(BASEURI, USERNAME, PASSWORD)
-    yesterday=date.today()-timedelta(days=1)
-    pjs=lims.get_projects(open_date=yesterday.strftime("%Y-%m-%d"))
-    for project in pjs:
-        if project.open_date:
-            project_ngi_identifier=project.id
-            project_ngi_name=project.name
+                    self.log.info('Updated internal id for order:{} - {}'.format(ORDER_ID, project.id))
+                else:
+                    print('{} Dry run: Updated internal id for order:{} - {}'.format(ORDER_ID, project.id))
+
+    def update_order_status(self, to_date, dry_run):
+        lims_db = get_session()
+        pjs = queries.get_last_modified_projectids(lims_db, "24 hours")
+        yesterday = (to_date-timedelta(days=1)).strftime("%Y-%m-%d")
+        today = to_date.strftime("%Y-%m-%d")
+        for p in pjs:
+            project = Project(self.lims, id=p)
             try:
                 ORDER_ID=project.udf['Portal ID']
             except KeyError:
                 continue
-            url = "{base}/api/v1/order/{id}".format(base=OP_BASE_URL, id=ORDER_ID)
-            data = {'fields': {'project_ngi_name': project.name, 'project_ngi_identifier': project.id}}
-            response = requests.post(url, headers=headers, json=data)
-            assert response.status_code == 200, (response.status_code, response.reason)
+            url = "{base}/api/v1/order/{id}".format(base=self.base_url, id=ORDER_ID)
+            response = requests.get(url, headers=self.headers)
+            data = ''
+            try:
+                data = response.json()
+            except ValueError: #In case a portal id does not exit on lims, skip the proj
+                continue
+            url = ''
+            status = ''
 
+            if (data['status'] == 'accepted' and
+                    project.udf.get('Queued') and
+                    not project.close_date):
+                url = data['links']['processing']['href']
+                status_set = 'processing'
+            if data['status'] == 'processing':
+                if project.udf.get('Aborted') and (project.udf.get('Aborted') == today or project.udf.get('Aborted') == yesterday):
```

Comment:
I'm not sure if the first `project.udf.get('Aborted')` is needed? Isn't this implied by the `== today or == yesterday` check? 🤔

Reply:
It throws an exception if it is null otherwise.

Reply:
Ah, makes sense 👍